### PR TITLE
Add support for the sharelink API

### PR DESCRIPTION
**Pull request by foreign-sub**
Add
create_share_link
delete_share_link
get_share_link
get_share_links

### CODE REVIEW

File: aiofreepybox/api/sharelink.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Sharelink:
+    """
+    Sharelink
+    """
+
+    def __init__(self, access):
+        self._access = access
+
+    share_link_data_schema = {"path": "", "expire": 0}
```
Comment (stilllman) [inline aiofreepybox/api/sharelink.py]:
```suggestion
    share_link_data_schema = {"path": "", "expire": 0, "fullurl": ""}
```

File: aiofreepybox/api/sharelink.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        token : `str`
+        """
+        await self._access.delete(f"share_link/{token}")
+
+    async def get_share_links(self):
+        """
+        Get share links
+        """
+        return await self._access.get("share_link/")
```
Comment (stilllman) [inline aiofreepybox/api/sharelink.py]: Add get_share_link(self, token) to call GET "share_link/{token}"

File: aiofreepybox/api/sharelink.py (hunk 1 above)
Comment (foreign-sub) [reply aiofreepybox/api/sharelink.py]: i've also changed it to take path and expire on input : create_share_link(self, path, expire)

File: aiofreepybox/api/sharelink.py (hunk 2 above)
Comment (foreign-sub) [reply aiofreepybox/api/sharelink.py]: fixed

**Pull request closed without merging**